iothub/service/src/RegistryManager.cs

```diff
@@ -31,6 +31,10 @@ public static RegistryManager CreateFromConnectionString(string connectionString
         /// <returns> An RegistryManager instance. </returns>
         public static RegistryManager CreateFromConnectionString(string connectionString, HttpTransportSettings transportSettings)
         {
+            if (transportSettings == null)
```

Comment:
this one is fine since the caller creates a new TransportSetting before calling into this method.